README.md

```diff
@@ -1,2 +1,494 @@
+![Logo](https://www.browserstack.com/images/static/header-logo.jpg)
+
+# BrowserStack Examples Cucumber TestNG <a href="https://cucumber.io"><img src="https://brandslogos.com/wp-content/uploads/images/large/cucumber-logo.png" alt="Cucumber" height="22" /></a> <a href="https://testng.org/"><img src="https://e7.pngegg.com/pngimages/640/776/png-clipart-testng-logo-software-testing-software-framework-computer-icons-automation-testing-angle-text.png" alt="TestNG" height="22" /></a>
+
+## Introduction
+
+TestNG is a testing framework designed to simplify a broad range of testing needs, from unit testing (testing a class in isolation of the others) to integration testing (testing entire systems made of several classes, several packages and even several external frameworks, such as application servers). Cucumber is a software tool that supports behavior-driven development (BDD).
+
+This BrowserStack Example repository demonstrates a Selenium test framework written in Cucumber and TestNG with parallel testing capabilties. The Selenium test scripts are written for the open source [BrowserStack Demo web application](https://bstackdemo.com) ([Github](https://github.com/browserstack/browserstack-demo-app)). This BrowserStack Demo App is an e-commerce web application which showcases multiple real-world user scenarios, written in Next and React. The app is bundled with offers data, orders data and products data that contains everything you need to start using the app and run tests out-of-the-box.
+
+The Selenium tests are run on different platforms like on-prem, docker and BrowserStack using various run configurations and test capabilities.
+
+
+
+---
+
+## Repository setup
+
+- Clone the repository
+
+- For this infrastructure configuration (i.e on-premise) ensure that the ChromeDriver executable is placed in the `/src/test/resources/` folder.
+
+- Ensure you have the following dependencies installed on the machine
+  - Java >= 8
+  - Maven >= 3.1+
+  - Gradle >= 5.0+
+
+  Maven:
+    ```sh
+    mvn install
+    ```
+
+  Gradle:
+    ```sh
+    gradle build
+    ```
+
+## About the tests in this repository
+
+This repository contains the following Selenium tests:
+
+| Module   | Test name                          | Description |
+  | ---   | ---                                   | --- |
+| E2E      | End to End Scenario                | This test scenario verifies successful product purchase lifecycle end-to-end. It demonstrates the [Page Object Model design pattern](https://www.browserstack.com/guide/page-object-model-in-selenium) and is also the default test executed in all the single test run profiles. |
+| Login    | Login with given username          | This test verifies the login workflow with different types of valid login users. |
```

Comment:
`mvn test -P on-prem -Dtest-name="Login with given username"`
does not trigger any test. pls check.